packages/arb-token-bridge-ui/src/hooks/useETHPrice.ts

```diff
@@ -11,21 +12,21 @@ export type UseETHPriceResult = {
 export function useETHPrice(): UseETHPriceResult {
   const [ethPrice, setETHPrice] = useState<ETHPrice>('loading')
 
-  useEffect(() => {
-    async function fetchETHPrice() {
-      try {
-        const response = await axios.get(
-          'https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd'
-        )
-
-        setETHPrice(response.data.ethereum.usd)
-      } catch (error) {
-        setETHPrice('error')
-      }
+  const { data, error } = useSWR(
+    'https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd',
+    url => axios.get(url).then(res => res.data),
+    {
+      refreshInterval: 30_000,
+      shouldRetryOnError: true,
+      errorRetryCount: 2,
+      errorRetryInterval: 3_000
     }
+  )
 
-    fetchETHPrice()
-  }, [])
+  useEffect(
+    () => setETHPrice(error ? 'error' : data?.ethereum?.usd),
```

Comment:
Or can we return `data` (as `ethPrice`) and `error` directly?

Reply:
Good point, completely forgot that SWR is a hook.

Also, I think it's better to return error directly, since price could change significantly between caches?